### PR TITLE
tusc-sh: init at 1.0.2

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

simple binary to run the tus protocol for file uploads.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/tusc-sh/default.nix
Comment (AndersonTorres) [inline pkgs/applications/networking/tusc-sh/default.nix:30]:
```suggestion
    description = "Tus 1.0.0 client protocol implementation for bash";
```

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:41221]: use pkgs/by-name

**Review by AndersonTorres — DISMISSED** [review]

**Comment by happysalada** [comment]
I took care of both comments.

File: pkgs/by-name/tu/tusc-sh/package.nix
Comment (mrcjkb) [inline pkgs/by-name/tu/tusc-sh/package.nix:23]:
```sh
mv tusc.sh tusc
install -Dm755 tusc -t $out/bin
```

This sets the appropriate permissions and creates any missing directories (no need for `mkdir` and `cp`).

**Review by mrcjkb — CHANGES_REQUESTED** [review]

**Comment by happysalada** [comment]
I've tested more extensively, thank you for suggesting!

I've resolved all the missing dependencies, and I have one error left, but it appears to be a protocol error.
I think this is good to merge right now. I'm working with upstream to try to understand where the issue could be.

File: pkgs/by-name/tu/tusc-sh/package.nix
Comment (mrcjkb) [inline pkgs/by-name/tu/tusc-sh/package.nix:28]: suggestion: now that you're using `writeShellApplication`, there's no need to rename `tusc.sh` to `tusc` (sorry this didn't occur to me before).

**Review by mrcjkb — APPROVED** [review]

File: pkgs/by-name/tu/tusc-sh/package.nix
Comment (happysalada) [reply pkgs/by-name/tu/tusc-sh/package.nix]: Thank you for the tip! Just pushed that change.

**Comment by happysalada** [comment]
After a brief interaction, upstream has made some very minor updates. Updated to 1.0.2
waiting for ofborg to pass before merging.
Let me know if anything though.

**Pull request merged**
